### ISSUE TITLE
Add informative prompt about "Never remember history" setting when database storage is unavailable

### ISSUE THREAD

**Issue by nandocova** [birchill/10ten-ja-reader#1507]
Hi! 

I'm on Firefox (Windows 10), latest updates. 10ten with the 1.17 version, and I can't use the Kanji and Names dictionaries. The regular definitions dictionary works fine when hovering over a word, but I can no longer see the tabs of the Kanji and Names dictionaries, no matter which ones are selected in the add-on options. The tabs don't appear in the pop-ups, and I can't access them with the key shortcuts (shift or enter) neither.

At the bottom of the add-ons options there's a message that reads: Database storage unavailable. Even if I hit the retry button next to it, nothing happens.

I tried re-installing the add-on with no success.

*edit: typo

**Comment by birtles** [comment]
Hi @nandocova, I'm really sorry to hear it isn't working for you. That sounds bad. Thank you for filing this.

I'm really not sure what could be causing that. The latest update (actually 1.16) adds a workaround for a Firefox bug that would cause the database to hang when changing languages but I don't _think_ it would cause it to say "Database storage unavailable".

I don't suppose you have enabled always-on private browsing mode or toggled any other privacy settings? Or installed other add-ons that might?

Firefox has a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1427986#c11) where Web pages and extensions can't use IndexedDB when always-on private browsing mode is enabled which would cause these symptoms.

**Comment by nandocova** [comment]
Hey @birtles ! 

Thanks for the reply, you nailed it! I had the "never remember history" option selected in my privacy and security settings. I changed it to custom settings and now 10ten is working as intended. The moment I restarted Firefox, the databases began updating and the dictionary tabs are back! 

Maybe a warning could be added in the add-on details for Firefox users? 

Anyways, thank you for your much appreciated help and work!
